lib/lambda-worker/lambda-worker.ts

```diff
@@ -0,0 +1,199 @@
+import * as eventSource from "@aws-cdk/aws-lambda-event-sources";
+import * as lambda from "@aws-cdk/aws-lambda";
+import * as lambdaNodeJs from "@aws-cdk/aws-lambda-nodejs";
+import * as sqs from "@aws-cdk/aws-sqs";
+import * as subs from "@aws-cdk/aws-sns-subscriptions";
+
+import { LambdaWorkerProps } from "./lambda-worker-props";
+
+const DEFAULT_MAX_RECEIVE_COUNT = 5;
+const DEFAULT_APPROX_AGE_OLDEST_MESSAGE_THRESHOLD = cdk.Duration.hours(1);
+const DEFAULT_APPROX_NUM_MESSAGES_VISIBLE_THRESHOLD = 1000;
+const MINIMUM_MEMORY_SIZE = 1024;
+const MINIMUM_LAMBDA_TIMEOUT = cdk.Duration.seconds(30);
+
+export class LambdaWorker extends cdk.Construct {
+  constructor(scope: cdk.Construct, id: string, props: LambdaWorkerProps) {
+    super(scope, id);
+
+    // Lambda settings
+    const memorySize =
+      props.lambdaProps.memorySize &&
+      props.lambdaProps.memorySize > MINIMUM_MEMORY_SIZE
+        ? props.lambdaProps.memorySize
+        : MINIMUM_MEMORY_SIZE;
+
+    const lambdaTimeout =
+      props.lambdaProps.timeout &&
+      props.lambdaProps.timeout.toSeconds() > MINIMUM_LAMBDA_TIMEOUT.toSeconds()
+        ? props.lambdaProps.timeout
+        : MINIMUM_LAMBDA_TIMEOUT;
+
+    // Queue settings
+    const maxReceiveCount = props.queueProps.maxReceiveCount
+      ? props.queueProps.maxReceiveCount
+      : DEFAULT_MAX_RECEIVE_COUNT;
+
+    const queueTimeout = cdk.Duration.seconds(
+      maxReceiveCount * lambdaTimeout.toSeconds()
+    );
+
+    const approximateAgeOfOldestMessageThreshold = props.queueProps
+      .approximateAgeOfOldestMessageThreshold
+      ? props.queueProps.approximateAgeOfOldestMessageThreshold
+      : DEFAULT_APPROX_AGE_OLDEST_MESSAGE_THRESHOLD;
+
+    const approximateNumberOfMessagesVisibleThreshold = props.queueProps
+      .approximateNumberOfMessagesVisibleThreshold
+      ? props.queueProps.approximateNumberOfMessagesVisibleThreshold
+      : DEFAULT_APPROX_NUM_MESSAGES_VISIBLE_THRESHOLD;
+
+    // Topic Subscription Settings
+    let subscriptionProps = {};
+    if (props.filterPolicy) {
+      subscriptionProps = { filterPolicy: props.filterPolicy };
+    }
+
+    // Create both the main queue and the dead letter queue
+    const lambdaDLQ = new sqs.Queue(this, `${props.name}-dlq`, {
+      queueName: `${props.name}-dlq`,
+      visibilityTimeout: queueTimeout,
+    });
+
+    const lambdaQueue = new sqs.Queue(this, `${props.name}-queue`, {
+      queueName: `${props.name}-queue`,
+      visibilityTimeout: queueTimeout,
+      deadLetterQueue: { queue: lambdaDLQ, maxReceiveCount: maxReceiveCount },
+    });
+
+    // If we have specified a topic, then subscribe
+    // the main queue to the topic.
+    if (props.topic) {
+      props.topic.addSubscription(
+        new subs.SqsSubscription(lambdaQueue, subscriptionProps)
+      );
+    }
+
+    // Create the lambda
+    const lambdaWorker = new lambdaNodeJs.NodejsFunction(
+      this,
+      `${props.name}`,
```

Comment:
I believe this is a string already, so you don't need to interpolate it:

```suggestion
      props.name,
```